snap-core/src/main/java/org/esa/snap/core/dataio/ProductIO.java

```diff
@@ -460,13 +471,44 @@ private static void writeAllBands(Product product, ProgressMonitor pm) throws IO
 
         if (!bandsToWrite.isEmpty()) {
             pm.beginTask("Writing bands of product '" + product.getName() + "'...", bandsToWrite.size());
+            final int numBands = bandsToWrite.size();
+            final int numThreads = Runtime.getRuntime().availableProcessors();
+            int executorSize = numThreads / numBands;
+            executorSize = executorSize == 0 ? 1 : executorSize;
+            Semaphore semaphore = null;
```

Comment:
This is needed because there is a slow down effect when there is only one executor which has to process a very long queue.
If you split up this executer, one executer per band, then there is no slow down.